app/jobs/partnership_activation_job.rb

```diff
@@ -0,0 +1,16 @@
+# frozen_string_literal: true
+
+class PartnershipActivationJob < ApplicationJob
+  discard_on ActiveJob::DeserializationError
+
+  def perform(partnership)
+    return if partnership.challenged?
+    return unless partnership.pending
+
+    ActiveRecord::Base.transaction do
+      partnership.update!(pending: false)
+      school_cohort = SchoolCohort.find_by!(school_id: partnership.school_id, cohort_id: partnership.cohort_id)
+      school_cohort.update!(induction_programme_choice: "full_induction_programme", core_induction_programme: nil)
```

Comment:
Still looks like it should be using the ObserverPattern to me.

Reply:
Still?

Reply:
By this you mean a single call to Partnership, which then updates the SchoolCohort? I don't think I agree